packages/components/src/components/domainproperties/models.ts

```diff
@@ -1285,27 +1292,30 @@ export class DomainFieldError extends Record({
     rowIndexes: List<number>;
     newRowIndexes?: List<number>;
 
-    static fromJS(rawFields: Array<any>, severityLevel: String): List<DomainFieldError> {
+    static fromJS(errors: Array<any>, severityLevel: String): List<DomainFieldError> {
 
         let fieldErrors = List<DomainFieldError>();
 
-        for (let i=0; i < rawFields.length; i++) {
+        let hasErrors = errors.find(error => error.severity === SEVERITY_LEVEL_ERROR);
+
+        for (let i=0; i < errors.length; i++) {
+
+            // stripping out server side warnings when there are errors
+            if (errors[i].id === SEVERITY_LEVEL_WARN && hasErrors)
+                continue;
 
             //empty field name and property id comes in as "form" string from the server, resetting it to undefined here
-            let fieldName = (rawFields[i].id === "form" && rawFields[i].field === "form" ? undefined : rawFields[i].field);
-            let propertyId = ((rawFields[i].id === "form" && rawFields[i].field === "form") || rawFields[i].id < 1 ? undefined : rawFields[i].id);
+            let fieldName = (errors[i].id === "form" && errors[i].field === "form" ? undefined : errors[i].field);
+            let propertyId = ((errors[i].id === "form" && errors[i].field === "form") || errors[i].id < 1 ? undefined : errors[i].id);
+            let severity = (errors[i].id === "Warning" ? SEVERITY_LEVEL_WARN : severityLevel);
 
-            let domainFieldError = new DomainFieldError({message: rawFields[i].message, fieldName, propertyId,
-                severity: severityLevel, serverError: true, rowIndexes: (rawFields[i].rowIndexes ? rawFields[i].rowIndexes : List<number>())});
+            let domainFieldError = new DomainFieldError({message: errors[i].message, fieldName, propertyId,
+                severity: severity, serverError: true, rowIndexes: (errors[i].rowIndexes ? errors[i].rowIndexes : List<number>())});
             fieldErrors = fieldErrors.push(domainFieldError);
         }
 
         return fieldErrors;
     }
-
-    constructor(values?: {[key:string]: any}) {
```

Comment:
Was there a specific reason for removing this? We have similar constructor functions for the rest of the models in this file.

Reply:
it got removed by mistake. thanks for catching this.